Mathlib/Topology/Category/ExtrDisc/Basic.lean

```diff
@@ -0,0 +1,177 @@
+
+namespace CompHaus
+
+/-- Proj implies ExtrDisc. -/
+@[simps!]
+def toExtrDisc (X : CompHaus.{u}) [Projective X] :
+    ExtrDisc where
+  compHaus := X
+  extrDisc := by
+    apply CompactT2.Projective.extremallyDisconnected
+    intro A B _ _ _ _ _ _ f g hf hg hsurj
+    have : CompactSpace (TopCat.of A) := by assumption
+    have : T2Space (TopCat.of A) := by assumption
+    have : CompactSpace (TopCat.of B) := by assumption
+    have : T2Space (TopCat.of B) := by assumption
+    let A' : CompHaus := ⟨TopCat.of A⟩
+    let B' : CompHaus := ⟨TopCat.of B⟩
+    let f' : X ⟶ B' := ⟨f, hf⟩
+    let g' : A' ⟶ B' := ⟨g,hg⟩
+    have : Epi g' := by
+      rw [CompHaus.epi_iff_surjective]
+      assumption
+    obtain ⟨h,hh⟩ := Projective.factors f' g'
+    refine ⟨h,h.2,?_⟩
+    ext t
+    apply_fun (fun e => e t) at hh
+    exact hh
+
+
+end CompHaus
+
+namespace ExtrDisc
+
+/-- Extremally disconnected spaces form a large category. -/
+instance : LargeCategory ExtrDisc.{u} :=
+  show Category (InducedCategory CompHaus (·.compHaus)) from inferInstance
+
+/-- The (forgetful) functor from extremally disconnected spaces to compact Hausdorff spaces. -/
+@[simps!]
+def toCompHaus : ExtrDisc.{u} ⥤ CompHaus.{u} :=
+  inducedFunctor _
+
+/-- Construct a term of `ExtrDisc` from a type endowed with the structure of a
+compact, Hausdorff and totally disconnected topological space.
+-/
+def of (X : Type _) [TopologicalSpace X] [CompactSpace X] [T2Space X]
+    [ExtremallyDisconnected X] : ExtrDisc :=
+  ⟨⟨⟨X, inferInstance⟩⟩⟩
+
+/-- The forgetful functor `ExtrDisc ⥤ CompHaus` is full. -/
+instance : Full toCompHaus where
+  preimage := fun f => f
+  witness := fun f => by simp
+
+
+/-- The forgetful functor `ExtrDisc ⥤ CompHaus` is faithful. -/
+instance : Faithful toCompHaus where
+  map_injective := by
+    intro X Y a b h
+    simp only [inducedFunctor_obj, inducedFunctor_map] at h
+    exact h
+
+/-- Extremally disconnected spaces are a concrete category. -/
+instance : ConcreteCategory ExtrDisc where
+  forget := toCompHaus ⋙ forget _
+
+instance : CoeSort ExtrDisc.{u} (Type u) := ConcreteCategory.hasCoeToSort _
+instance {X Y : ExtrDisc.{u}} : FunLike (X ⟶ Y) X (fun _ => Y) := ConcreteCategory.funLike
+
+/-- Extremally disconnected spaces are topological spaces. -/
+instance instTopologicalSpace (X : ExtrDisc.{u}) : TopologicalSpace X :=
+  show TopologicalSpace X.compHaus from inferInstance
+
+/-- Extremally disconnected spaces are compact. -/
+instance (X : ExtrDisc.{u}) : CompactSpace X :=
+  show CompactSpace X.compHaus from inferInstance
+
+/-- Extremally disconnected spaces are Hausdorff. -/
+instance (X : ExtrDisc.{u}) : T2Space X :=
+  show T2Space X.compHaus from inferInstance
+
+instance (X : ExtrDisc.{u}) : ExtremallyDisconnected X :=
+  X.2
+
+/-- Extremally disconnected spaces are totally separated. -/
+instance (X : ExtrDisc.{u}) : TotallySeparatedSpace X :=
+{ isTotallySeparated_univ := by
```

Comment:
We should add a suitable instance for topological spaces, and use that instance here.